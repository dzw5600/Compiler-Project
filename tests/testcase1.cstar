// Basic tests:
// 1. Declaration of variables (assignment immediately, and reassignment using `set`)
// 2. Addition, multiplication, modulus
// 3. Print statements

// Addition
int a = 5;
int b;
set b = a + 10;
print(a); // 5
print(b); // 15

// Multiplication
int c = 10;
int mult = c * b;
print(mult); // 150

// Modulus
set a = 4; // Reassignment of already non-null value
int mod = a % 4;
print(mod); // 0