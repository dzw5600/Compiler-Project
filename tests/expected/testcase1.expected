5
15
150
0